lib/Doctrine/DBAL/Migrations/Version.php

```diff
@@ -256,12 +257,21 @@ public function execute($direction, $dryRun = false)
             if (! $dryRun) {
                 if ($this->sql) {
                     foreach ($this->sql as $key => $query) {
-                        if ( ! isset($this->params[$key])) {
-                            $this->outputWriter->write('     <comment>-></comment> ' . $query);
-                            $this->connection->executeQuery($query);
-                        } else {
-                            $this->outputWriter->write(sprintf('    <comment>-</comment> %s (with parameters)', $query));
-                            $this->connection->executeQuery($query, $this->params[$key], $this->types[$key]);
+                        try {
+                            if ( ! isset($this->params[$key])) {
+                                $this->outputWriter->write('     <comment>-></comment> ' . $query);
+                                $this->connection->executeQuery($query);
+                            } else {
+                                $this->outputWriter->write(sprintf('    <comment>-</comment> %s (with parameters)', $query));
+                                $this->connection->executeQuery($query, $this->params[$key], $this->types[$key]);
+                            }
+                        } catch (DBALException $e) {
+                            if ($continueAfterErrors) {
+                                $this->outputWriter->write(sprintf('         <error>- failed</error>', $query));
```

Comment:
the output message should include the exception message.

and your sprintf with the SQL is useless because there is not placeholder in your string (and there is no reason to put the SQL again here because it has already been displayed before running the query)